hls4ml/model/hls_layers.py

```diff
@@ -448,15 +451,17 @@ def make_stream_variable(self, shape, dim_names, var_name='layer{index}_out', ty
         
         return StreamVariable(shape, dim_names, var_name=var_name, type_name=type_name, precision=precision, n_pack=pack_factor, depth=depth, index=self.index)
 
-    def add_weights(self, quantizer=None, compression=False):
-        data = self.model.get_weights_data(self.name, 'kernel')
+    def add_weights(self, quantizer=None, compression=False, data=None):
```

Comment:
After making below change `Conv2DBatchnorm(Layer)` -> `Conv2DBatchnorm(Conv2D)`, you won't need these any more.
`add_weights_variable` already has the functionality to add the data.